File: Python/1094.py
cobaia=0
coelho=0
rato=0
sapo=0

n=int(input())
for i in range(n):
    linha=input().split(" ")
    x=int(linha[0])
    animal=(linha[1])
    if (animal=='C'):
        coelho+=x
    elif (animal=='S'):
        sapo+=x
    elif (animal=='R'):
        rato+=x
    cobaia+=x
    
print("Total: {} cobaias" .format(cobaia))
print("Total de coelhos: {}" .format(coelho))
print("Total de ratos: {}" .format(rato))
print("Total de sapos: {}" .format(sapo))
print("Percentual de coelhos: {:.2f} %" .format((coelho*100)/cobaia))
print("Percentual de ratos: {:.2f} %" .format((rato*100)/cobaia))
print("Percentual de sapos: {:.2f} %" .format((sapo*100)/cobaia))

File: C/1094.c
#include <stdio.h>
#include <string.h>
    int main() {
        int i, sapo=0, coelho=0, rato=0, n, entrada, total=0; 
        char animal[1];
        double pc, pr,ps;
        scanf("%d", &n);
        for (i=0; i<n; i++) {
            scanf("%d %c", &entrada, animal);
            if (strcmp(animal, "C")==0) coelho+=entrada*100;
            else if (strcmp(animal, "R")==0) rato+=entrada*100;
            else if (strcmp(animal, "S")==0) sapo+=entrada*100;
        }

        total=sapo+coelho+rato;
        printf("Total: %d cobaias\n", total/100);
        printf("Total de coelhos: %d\n", coelho/100);
        printf("Total de ratos: %d\n", rato/100);
        printf("Total de sapos: %d\n", sapo/100);
        pc=(100.0*coelho)/(total);
        pr=(100.0*rato)/(total);
        ps=(100.0*sapo)/(total);
        if (n==0) {
        printf("Percentual de coelhos: 0.00%%\n");
        printf("Percentual de ratos: 0.00%%\n");
        printf("Percentual de sapos: 0.00%%\n");
        } else {
        printf("Percentual de coelhos: %.2lf %%\n", pc);
        printf("Percentual de ratos: %.2lf %%\n", pr);
        printf("Percentual de sapos: %.2lf %%\n", ps);
        }
        return 0;
    }


